js/src/components/paid-ads/asset-group/texts-editor.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import GridiconCrossSmall from 'gridicons/dist/cross-small';
+
+/**
+ * Internal dependencies
+ */
+import AppButton from '.~/components/app-button';
+import AppInputControl from '.~/components/app-input-control';
+import AddAssetItemButton from './add-asset-item-button';
+import './texts-editor.scss';
+
+/**
+ * Renders a list of text inputs for managing the single type of asset texts.
+ *
+ * @param {Object} props React props.
+ * @param {string[]} [props.initialTexts=[]] Initial texts.
+ * @param {number} [props.minNumberOfTexts=0] Minimum number of texts.
+ * @param {number} [props.maxNumberOfTexts=0] Maximum number of texts.
+ * @param {number|number[]} props.maxCharacterCounts Maximum number of characters for each text. If the limits are the same, a single number can be used instead of an array.
+ * @param {string} props.addButtonText Text for the button to add a new text input.
+ * @param {string} [props.placeholder] Placeholder text.
+ * @param {Function} props.onChange Callback function to be called when the texts are changed.
+ */
+export default function TextsEditor( {
+	initialTexts = [],
+	minNumberOfTexts = 0,
+	maxNumberOfTexts = 0,
+	maxCharacterCounts,
+	addButtonText,
+	placeholder,
+	onChange,
+} ) {
+	const [ texts, setTexts ] = useState( () => {
+		const shortage = minNumberOfTexts - initialTexts.length;
+		if ( shortage <= 0 ) {
+			return initialTexts;
+		}
```

Comment:
💅  This could be committed since `Array.from( { length: shortage }, () => '' );` with  length <= 0 will return an empty array.

Reply:
Removed along with the change in https://github.com/woocommerce/google-listings-and-ads/pull/1854/commits/0fe5fc9bc716542590ac57d3f06c7c9dce6b9470.

The reason was that calling `Array.from` with a negative `length` looks semantically buggy. Therefore, I added a `Math.max()`.